step-templates/sitecore-ship-deploy.json

```diff
@@ -0,0 +1,82 @@
\t}\r\n        $packageFileStream = New-Object System.IO.FileStream @($InFile, [System.IO.FileMode]::Open)\r\n        \r\n\t\t$contentDispositionHeaderValue = New-Object System.Net.Http.Headers.ContentDispositionHeaderValue \"form-data\"\r\n\t    $contentDispositionHeaderValue.Name = \"fileData\"\r\n\t\t$contentDispositionHeaderValue.FileName = (Split-Path $InFile -leaf)\r\n \r\n        $streamContent = New-Object System.Net.Http.StreamContent $packageFileStream\r\n        $streamContent.Headers.ContentDisposition = $contentDispositionHeaderValue\r\n        $streamContent.Headers.ContentType = New-Object System.Net.Http.Headers.MediaTypeHeaderValue $ContentType\r\n        \r\n        $content = New-Object System.Net.Http.MultipartFormDataContent\r\n        $content.Add($streamContent)\r\n \r\n        try\r\n        {\r\n\t\t\t$response = $httpClient.PostAsync($Uri, $content).Result\r\n \r\n\t\t\tif (!$response.IsSuccessStatusCode)\r\n\t\t\t{\r\n\t\t\t\t$responseBody = $response.Content.ReadAsStringAsync().Result\r\n\t\t\t\t$errorMessage = \"Status code {0}. Reason {1}. Server reported the following message: {2}.\" -f $response.StatusCode, $response.ReasonPhrase, $responseBody\r\n \r\n\t\t\t\tthrow [System.Net.Http.HttpRequestException] $errorMessage\r\n\t\t\t}\r\n \r\n\t\t\treturn $response.Content.ReadAsStringAsync().Result\r\n        }\r\n        catch [Exception]\r\n        {\r\n\t\t\t$PSCmdlet.ThrowTerminatingError($_)\r\n        }\r\n        finally\r\n        {\r\n            if($null -ne $httpClient)\r\n            {\r\n                $httpClient.Dispose()\r\n            }\r\n \r\n            if($null -ne $response)\r\n            {\r\n                $response.Dispose()\r\n            }\r\n        }\r\n    }\r\n    END { }\r\n}\r\n<#Invoke-MultipartFormDataUpload#>\r\n\r\n\r\n$fileUploadUrl = \"$SiteUrl/services/package/install/fileupload\"\r\n\r\nif($Username){\r\n\t$secpasswd = ConvertTo-SecureString $Password -AsPlainText -Force\r\n\t$mycreds = New-Object System.Management.Automation.PSCredential ($Username, $secpasswd)\r\n\tWrite-Host \"POST Fileupload with credentials\" -foregroundcolor green\r\n\t$shipResponse = Invoke-MultipartFormDataUpload -InFile $UpdatePackagePathZip -Uri $fileUploadUrl -Timeout $ConnectionTimeOutInSeconds -Credential $mycreds | ConvertFrom-Json\r\n}\r\nelse{\r\n\tWrite-Host \"POST Fileupload\" -foregroundcolor green\r\n\t$shipResponse = Invoke-MultipartFormDataUpload -InFile $UpdatePackagePathZip -Uri $fileUploadUrl -Timeout $ConnectionTimeOutInSeconds  | ConvertFrom-Json\r\n}\r\n\r\n$shipEntries = $shipResponse.Entries\r\n$items = New-Object System.Collections.Generic.List[string]\r\n$shipEntries | \r\n\tForEach-Object {\r\n\t\tif($_.ID -ne $null)\r\n\t\t{\r\n\t\t\tWrite-Host $_.ID -foregroundcolor cyan\r\n\t\t\t$items.Add($_.ID )\r\n\t\t}\r\n\t}\r\n\t\r\nif($ResultsOutputPath -ne $null)\r\n{\r\n    Write-Host \"Saving Results to:\" $ResultsOutputPath\r\n\t$items  | out-file $ResultsOutputPath\r\n}\r\n\r\n",
```

Comment:
```
if($ResultsOutputPath -ne $null)
{
    Write-Host \"Saving Results to:\" $ResultsOutputPath
	$items  | out-file $ResultsOutputPath
}
```
I think a nice addition for this would be to use the `New-OctopusArtifact` helper method to attach this file as an artifact of the release, e.g. `New-OctopusArtifact -Path $ResultsOutputPath -Name "Sitecore.Ship_Deploy.Results.txt"`